subsys/bluetooth/host/conn.c

```diff
@@ -1732,9 +1737,18 @@ static void perform_auto_initiated_procedures(struct bt_conn *conn, void *unused
 
 	if (!atomic_test_bit(conn->flags, BT_CONN_LE_FEATURES_EXCHANGED) &&
 	    can_initiate_feature_exchange(conn)) {
-		err = bt_hci_le_read_remote_features(conn);
-		if (err) {
-			LOG_ERR("Failed read remote features (%d)", err);
+
+		if (IS_ENABLED(CONFIG_BT_LL_EXTENDED_FEAT_SET) &&
+		    BT_FEAT_LE_LL_EXTENDED_FEAT_SET(bt_dev.le.features)) {
+			err = bt_hci_le_read_all_remote_features(conn);
```

Comment:
Reading all features may potentially take quite some time, delaying other setup. If `CONFIG_BT_LE_MAX_SUPPORTED_FEATURE_PAGE` is set to a large value, we could end up degrading user experience with this.

Therefore, I suggest keeping it as is. That is, continue to use `bt_hci_le_read_remote_features`.

We could even consider making this configurable as well - there is really no need to know the peer feature set. The controller will handle this anyhow.

Reply:
I have updated so the procedure no longer triggers automatically, only on command. Also we do not store extended features for the remote device, only for the local features